### PR TITLE
Remove worker-rt MCP with deploy clean-up

### PR DESCRIPTION
**Pull request by gsr-shanks**
Remove worker-rt MCP while performing 'make cluster-clean'

### CODE REVIEW

File: hack/clean-deploy.sh [hunk 1]
```diff
@@ -7,4 +7,6 @@ $OC_TOOL delete performanceprofile --all
 
 $OC_TOOL delete ns openshift-performance-addon
 
-$OC_TOOL -n openshift-marketplace delete catalogsource performance-addon-operator-catalogsource
\ No newline at end of file
+$OC_TOOL -n openshift-marketplace delete catalogsource performance-addon-operator-catalogsource
+
+$OC_TOOL delete mcp worker-rt
```
Comment (slintes) [inline hack/clean-deploy.sh]: missing new line at the end

Comment (gsr-shanks) [reply hack/clean-deploy.sh]: Done.

**Comment by cynepco3hahue** [comment]
@gsr-shanks Removing the MCP before you remove the role from the node will drop the node to the degraded state.
Once you remove MCP, the system will remove rendered MC, so machine config controller will fail to find the current configuration and will fail to rollback node to default worker MCP.

IMHO the right way for clean up:
- remove `worker-rt` role from the node
- wait until the node appear back under the worker MCP
- wait for worker MCP updated status
- remove the worker-rt MCP

**Comment by slintes** [comment]
> @gsr-shanks Removing the MCP before you remove the role from the node will drop the node to the degraded state.
> Once you remove MCP, the system will remove rendered MC, so machine config controller will fail to find the current configuration and will fail to rollback node to default worker MCP.
> 
> IMHO the right way for clean up:
> 
> * remove `worker-rt` role from the node
> * wait until the node appear back under the worker MCP
> * wait for worker MCP updated status
> * remove the worker-rt MCP

oh wow, didn't know that.
sounds like a MCO bug to me
At least I would expect that MCO just switches the node to worker MCP when worker-rt is removed

**Comment by cynepco3hahue** [comment]
@slintes Yep, I still need to open one, just wanted to have the appropriate error message from logs, will do it today.

**Comment by slintes** [comment]
superseded by #67 

/close

**Pull request closed without merging**